File: python/roll_dice_sum.py
# https://www.codewars.com/kata/56f78a42f749ba513b00037f/solutions

from itertools import product

def rolldice_sum_prob(sum_, dice_amount):
    dice_values = [1,2,3,4,5,6]
    counter = 0
    for c in product(dice_values, repeat = dice_amount):
        counter += 1 if sum(list(c)) == sum_ else 0
    prob = counter/6**dice_amount
    return prob             

File: go/roll_dice_sum.go
//https://www.codewars.com/kata/56f78a42f749ba513b00037f/train/go

package kata

//Bernoulli Distribution
func RolldiceSumProb(sum, dice int) float64 {
  if dice == 0 && sum == 0 {
    return 1
  } else if sum < dice || sum > 6 * dice {
    return 0
  } else {
    sumProb := 0.0
    for i := 1; i <= 6; i++ {
      sumProb += RolldiceSumProb(sum - i, dice - 1)
    }
    return sumProb / 6
  }
}
